### PR TITLE
Fix MapboxMap extension plugin functions throwing exceptions

### PR DESCRIPTION
**Pull request by kiryldz**
### Summary of changes

When using extension functions for camera and gesture plugins (e.g. `MapboxMap.flyTo` and others) we previously were relying on weak references. That could lead to runtime crash:
```kotlin
checkNotNull(cameraAnimationsPlugin?.get()) {
  "Camera plugin is not added as the part of MapInitOptions for given MapView."
}
```
not only when camera plugin was excluded (as was per design) but also let's say when `MapView` was already destroyed and extension function was accessed.

Fixed this by removing weak references and improving warning messages.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 1]
```diff
@@ -1699,9 +1698,10 @@ class MapboxMap :
   }
 
   internal fun setCameraAnimationPlugin(cameraAnimationsPlugin: CameraAnimationsPlugin?) {
-    cameraAnimationsPlugin?.let {
-      if (it is CameraAnimationsPluginImpl)
-        this.cameraAnimationsPlugin = WeakReference(it)
+    if (cameraAnimationsPlugin == null || cameraAnimationsPlugin is CameraAnimationsPluginImpl) {
+      this.cameraAnimationsPlugin = cameraAnimationsPlugin
+    } else {
+      logW(TAG, "MapboxMap camera extension functions could work only with Mapbox developed plugin!")
```
[the remark this answers is not in the document]
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1704]: classed in Kotlin are final by default, aren't they? If we want to make them non-final, we should add `open` afair.

File: sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt [hunk 2]
```diff
@@ -969,20 +969,40 @@ class MapboxMapTest {
 
   @Test
   fun setCameraAnimationsPluginInvalid() {
+    mockkStatic("com.mapbox.maps.MapboxLogger")
+    every { logW(any(), any()) } just Runs
     mapboxMap.setCameraAnimationPlugin(mockk())
     assertNull(mapboxMap.cameraAnimationsPlugin)
+    unmockkStatic("com.mapbox.maps.MapboxLogger")
```
[the remark this answers is not in the document]
Comment (kiryldz) [reply sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt:976]: we're not doing this in similar cases and imo it's not useful

[the remark this answers is not in the document]
Comment (alexander-kulikovskii) [reply sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt:976]: we can start :slightly_smiling_face: And I don't think that it useful. You print some log for our users.

[the remark this answers is not in the document]
Comment (kiryldz) [reply sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt:976]: I think it was discussed in some other PR and validating log messages is a bit of an overkill. It's enough to know that some instance is null for the business logic

**Review by alexander-kulikovskii — APPROVED** [review]

**Review by Chaoba — APPROVED** [review]

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt [hunk 3]
```diff
@@ -15,132 +15,116 @@ val MapPluginProviderDelegate.camera: CameraAnimationsPlugin
...
+ * `MapView` as part of `MapInitOptions.plugins`.
  *
  * @param cameraOptions The camera options to ease to
  * @param animationOptions Transition options (animation duration, listeners etc)
  *
- * @return [Cancelable] animator set object or null if associated map object was garbage collected.
- *
- * @throws IllegalStateException if gestures plugin was not added.
- * @see [com.mapbox.maps.MapInitOptions]
+ * @return [Cancelable] animator set object or empty cancelable object if associated map object was garbage collected.
```
Comment (yunikkk) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt:30]:
```suggestion
 * @return [Cancelable] animator set object or empty cancelable object if associated MapView was destroyed.
```

Comment (yunikkk) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt:30]: And other places as well

File: sdk/src/main/java/com/mapbox/maps/MapController.kt [hunk 4]
```diff
@@ -169,6 +169,8 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
       return
     }
     lifecycleState = LifecycleState.STATE_DESTROYED
+    mapboxMap.setCameraAnimationPlugin(null)
```
Comment (yunikkk) [inline sdk/src/main/java/com/mapbox/maps/MapController.kt:172]: Could it actually be reset in mapboxMap.onDestroy instead?

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt (hunk 3 above)
Comment (kiryldz) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt:30]: yep

File: sdk/src/main/java/com/mapbox/maps/MapController.kt (hunk 4 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapController.kt:172]: 💎

**Pull request merged**